### PR TITLE
Level complete script

### PR DESCRIPTION
**Pull request by AlexBowmanCoding**
made a script that you attach to an area2d and use the exported variable to asign a scene to change to

### CODE REVIEW

File: level_complete.gd [hunk 1]
```diff
@@ -0,0 +1,8 @@
+extends Area2D
+export(String, FILE) var next_level = ""
+
+
+
+
+func _on_Area2D_body_entered(body):
+	get_tree().change_scene(next_level)
```
Comment (laylabmanley) [inline level_complete.gd:8]: Please rework to use functions provided by: https://github.com/Boxnet-Media/boxnetjam/pull/41

e.g. `global.next_level()`

**Review by laylabmanley — CHANGES_REQUESTED** [review]

**Comment by laylabmanley** [comment]
Also this does not include `If there area no more levels, the player should be brought to a victory scene.`

You can use the signal `completed_all_levels` on the `global` object to be alerted when the last level is completed.

**Review by laylabmanley — CHANGES_REQUESTED** [review]
Great changes! Two things:

1. you need to call on _ready connect(self, "on_body_entered", "_on_Area2D_body_entered")

File: level_complete.gd [hunk 2]
```diff
@@ -0,0 +1,11 @@
+extends Area2D
+export var last_level = ""
```
Comment (laylabmanley) [inline level_complete.gd:2]: change this to `export(PackedScene) var last_level`

**Review by laylabmanley — CHANGES_REQUESTED** [review]

File: level_complete.gd (hunk 2 above)
Comment (laylabmanley) [reply level_complete.gd:2]: Actually, this should be a preload and be a victory scene which you'll have to create

Comment (laylabmanley) [reply level_complete.gd:2]: So steps:
- [ ] Create victory scene
- [ ] Save it
- [ ] Get resource path
- [ ] `var last_level = preload(NODE_PATH_GOES_HERE)`

Comment (AlexBowmanCoding) [reply level_complete.gd]: Oh okay

File: level_complete.gd [hunk 3]
```diff
@@ -0,0 +1,14 @@
+extends Area2D
+export var last_level = "res://Scenes/victory!.tscn"
```
Comment (laylabmanley) [inline level_complete.gd:2]: needs to be a preload

e.g. `preload("res://Scenes/victory!.tscn)`

Also, the tscn needs to be included AND should not have special characters in the name.

**Review by laylabmanley — CHANGES_REQUESTED** [review]

File: level_complete.gd [hunk 4]
```diff
@@ -0,0 +1,16 @@
+extends Area2D
+
+export var last_level = "res://Scenes/victory.tscn"
+
+
+
+
+func _ready():
+	self.connect( "on_body_entered", self, "_on_Area2D_body_entered")
+	preload("res://Scenes/victory.tscn")
```
Comment (laylabmanley) [inline level_complete.gd:10]: This line doesn't do anything

**Review by laylabmanley — CHANGES_REQUESTED** [review]

**Review by laylabmanley — APPROVED** [review]

**Pull request merged**